protocol/network/4-architecture.md

```diff
@@ -0,0 +1,126 @@
+││ kad-routing ┌──────────────┐┌──────────────┐┌──────────────┐ ││
+││             │mDNS-discovery││random-walk   ││bootstrap-list│ ││
+││             └──────────────┘└──────────────┘└──────────────┘ ││
+│└──────────────────────────────────────────────────────────────┘│
+│┌──────────────────────────────────────────────────────────────┐│
+││ mDNS-routing                                                 ││
+│└──────────────────────────────────────────────────────────────┘│
+│┌──────────────────────────────────────────────────────────────┐│
+││ other-routing-mechanisms                                     ││
+│└──────────────────────────────────────────────────────────────┘│
+└────────────────────────────────────────────────────────────────┘
+```
+
+#### 4.1.1.1 mDNS-discovery
+
+mDNS-discovery uses mDNS to emit beacons in the local area network to find if there are more peers available, these peers are extremely interesting because our link to them will have an low latency, plus if we are disconnected from the network, we will be leverage the content on those peers have cached.
+
+#### 4.1.1.2 random-walk
+
+Random walking consists on traversing the routing tables of the peers available in the network, learning about new peers along the way.
+
+#### 4.1.1.3 bootstrap-list
+
+A predefined set of peers available on the network in well known locations, so that a new created node can easily find more peers to connect in the network.
+
+### 4.1.2 mDNS-routing
+
+mDNS-routing uses mDNS probes to identify if local area network peers that have a given key or simply are present.
+
+
+
+
+
+
+
+
+## 4.2 Swarm
+
+### 4.2.1 Stream Muxer
+
+The stream muxer must implement the interface offered by [abstract-stream-muxer](https://github.com/diasdavid/abstract-stream-muxer).
+
+### 4.2.2 Protocol Muxer
+
+Protocol muxing is handled on the application level instead of the conventional way at the port level (where a different services/protocols listen at different ports). This enables us to support several protocols to be muxed in the same socket, saving the cost of doing NAT traversal for more than one port.
+
+Protocol multiplexing is done through [`multistream`](https://github.com/jbenet/multistream), a protocol to negoatiate different types of streams (protocols) using [`multicodec`](https://github.com/jbenet/multicodec)
+
+### 4.2.3 Transport
+
+### 4.2.4 Crypto
+
+### 4.2.5 Identify
+
+**Identify** is one of the protocols mounted on top of swarm, our Connection handler, however, it follows and respects the same pattern as any other protocol when it comes to mounting it on top of swarm. Identify enables us to trade listenAddrs and observedAddrs between peers, this is crucial for the working of IPFS, since every socket open implements REUSEPORT, an observedAddr by another peer can enable a third peer to connect to us, since the port will be already open and redirect to us on a NAT.
+
+### 4.2.6 Relay
```

Comment:
i think we can collect `{Identify, Relay, NAT (there's multiple sub protocols for NAT), etc}` into a heading like `"Connectivity Protocols"`, which layer over Swarm, but are part of (under) libp2p (`libp2p.Node`).